src/app/form/_utils/DataStoreProvider.tsx

```diff
@@ -10,6 +11,22 @@ export const DataStoreProvider = ({ children }: { children: React.ReactNode }) =
     setDataStore({ ...dataStore, ...dataUpdates });
   };
 
+  const hasData = Object.keys(dataStore).length > 0;
+  useEffect(() => {
+    const handleBeforeUnload = (event: BeforeUnloadEvent) => {
+      if (hasData) {
+        event.preventDefault();
+        event.returnValue = true; // legacy browser support
```

Comment:
[dust] Specifying which ones might help someone decide in the future whether it's safe to remove this